snapcraft/internal/sources/errors.py

```diff
@@ -23,3 +23,13 @@ class IncompatibleOptionsError(errors.SnapcraftError):
 
     def __init__(self, message):
         super().__init__(message=message)
+
+
+class DigestDoesNotMatchError(errors.SnapcraftError):
```

Comment:
I remember writing special exceptions, didn't know we got a new error interface since I initially wrote this, cool!